store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow.md

```diff
@@ -256,6 +302,58 @@ After a successful file upload the logs should look something like:
+
+Once both checks pass, you know the file is committed on-chain and the MSP backend is ready to serve it, so the subsequent download call won’t randomly fail with a `404` while the system is still syncing.
+
+1. Add the following code in your `fileOperations.ts` file:
+    
+    ```ts title="fileOperations.ts"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/fileOperations.ts:wait-for-msp-confirm-on-chain'
+
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/fileOperations.ts:wait-for-backend-file-ready'
+    ```
+
+2. Update the `index.ts` file to trigger the helper method you just implemented:
+
+    ```ts title="index.ts // **PLACEHOLDER FOR STEP 6: WAIT FOR BACKEND TO HAVE FILE**"
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:wait-for-msp-confirm-on-chain'
+
+    --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:wait-for-backend-file-ready'
+    ```
+
+    ??? code "View complete `index.ts` file up until this point"
+
+        ```ts title="src/index.ts"
+        --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:imports'
+
+        async function run() {
+        // For anything from @storagehub-sdk/core to work, initWasm() is required
+        // on top of the file
+        await initWasm();
+        
+        // --- End-to-end storage flow ---
+        --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:check-msp-health'
+        --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:create-bucket'
+        --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:verify-bucket'
+        --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:wait-for-backend-bucket-ready'
+        --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:upload-file'
+        --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/end-to-end-storage-workflow/end-to-end-storage-workflow.ts:wait-for-backend-file-ready'
+        // **PLACEHOLDER FOR STEP 7: DOWNLOAD FILE**
+        // **PLACEHOLDER FOR STEP 8: VERIFY FILE**
+
+        // Disconnect the Polkadot API at the very end
+        await polkadotApi.disconnect();
+        }
+
+        await run();
+        ```
+
 ## Download and Save File
 
 Download the file by its deterministic key from the MSP and save it locally.
```

Comment:
Seeing this now, it just looks a little weird haha. Can you rephrase this to say something like "In the following sections, you will download the file..."?